Azure/azure-cli
# [App Service] Fix: Surface error when invalid webapp name provided

**Description**  
Closes #11884 

Show descriptive error when providing an invalid name in `az webapp up -n appName`.
Currently, providing an invalid name (for example, a name with an underscore) shows the error 'Unable to retrieve details of the existing app {invalid_name}. Please check that the app is a part of the current subscription', even though there is no existing app. 

**Testing Guide**  
Run `az webapp up -n invalid_name` in the directory of a test app you want to deploy. The error should now be more descriptive of the problem, and show 'Hostname must be less than 64 chars and must follow allowed character guidelines at http://tools.ietf.org/html/rfc952'. 

All other functionality of the command should stay the same. 

**History Notes**  


---

This checklist is used to make sure that common guidelines for a pull request are followed.
